GitUI/NBugReports/BugReporter.cs

```diff
@@ -18,123 +18,119 @@ public static class BugReporter
+        /// </summary>
+        /// <param name="text">A StringBuilder to which the exception data is appended.</param>
+        /// <param name="exception">An Exception to describe.</param>
+        /// <returns>The inner-most exception message.</returns>
+        internal static string Append(StringBuilder text, Exception exception)
         {
-            StringBuilder sb = new();
-
-            // Command: <command>
-            if (!string.IsNullOrWhiteSpace(exception.Command))
+            string rootError = exception.Message;
+            for (Exception innerException = exception.InnerException; innerException is not null; innerException = innerException.InnerException)
             {
-                sb.AppendLine($"{Strings.Command}: {exception.Command}");
+                if (!string.IsNullOrEmpty(innerException.Message))
+                {
+                    rootError = innerException.Message;
+                }
             }
 
-            // Arguments: <args>
-            if (!string.IsNullOrWhiteSpace(exception.Arguments))
+            if (exception is UserExternalOperationException userExternalOperationException && !string.IsNullOrWhiteSpace(userExternalOperationException.Context))
             {
-                sb.AppendLine($"{Strings.Arguments}: {exception.Arguments}");
+                // Context contains an error message as UserExternalOperationException is currently used. So append just "<context>"
+                text.AppendLine(userExternalOperationException.Context);
             }
 
-            // Working directory: <working dir>
-            sb.AppendLine($"{Strings.WorkingDirectory}: {exception.WorkingDirectory}");
-
-            if (canRaiseBug)
+            if (exception is ExternalOperationException externalOperationException)
             {
-                // Directions to raise a bug
-                sb.AppendLine();
-                sb.AppendLine(Strings.ReportBug);
-            }
+                // Command: <command>
+                AppendUnlessEmpty(externalOperationException.Command, GitUI.Strings.Command);
 
-            return sb.ToString();
-        }
+                // Arguments: <args>
+                AppendUnlessEmpty(externalOperationException.Arguments, GitUI.Strings.Arguments);
 
-        public static void Report(Exception exception, bool isTerminating)
-        {
-            if (exception is UserExternalOperationException userExternalException)
-            {
-                // Something happened that was likely caused by the user
-                ReportUserException(userExternalException, isTerminating);
-                return;
+                // Directory: <dir>
+                AppendUnlessEmpty(externalOperationException.WorkingDirectory, GitUI.Strings.WorkingDirectory);
             }
 
-            if (exception is ExternalOperationException externalException)
+            return rootError;
+
+            void AppendUnlessEmpty(string? value, string designation)
```

Comment:
Suggest to align with the established `IfNot` conventions in our codebase:
```suggestion
            void AppendIfNotEmpty(string? value, string designation)
```
or
```suggestion
            void AppendIfPresent(string? value, string designation)
```

Reply:
Renamed.
I will squash-merge tomorrow.